packages/language-service/lib/index.js

```diff
@@ -0,0 +1,891 @@
+/**
+ * @typedef {import('mdast').Root} Root
+ * @typedef {import('typescript').Diagnostic} Diagnostic
+ * @typedef {import('typescript').DiagnosticWithLocation} DiagnosticWithLocation
+ * @typedef {import('typescript').DocumentSpan} DocumentSpan
+ * @typedef {import('typescript').IScriptSnapshot} IScriptSnapshot
+ * @typedef {import('typescript').LanguageService} LanguageService
+ * @typedef {import('typescript').LanguageServiceHost} LanguageServiceHost
+ * @typedef {import('typescript').NavigationBarItem} NavigationBarItem
+ * @typedef {import('typescript').SymbolDisplayPart} SymbolDisplayPart
+ * @typedef {import('typescript').TextSpan} TextSpan
+ * @typedef {import('unified').PluggableList} PluggableList
+ * @typedef {import('./utils.js').MDXSnapshot} MDXSnapshot
+ */
+
+import remarkMdx from 'remark-mdx'
+import remarkParse from 'remark-parse'
+import {unified} from 'unified'
+
+import {toDiagnostic} from './error.js'
+import {getMarkdownDefinitionAtPosition} from './markdown.js'
+import {bindAll} from './object.js'
+import {getFoldingRegions} from './outline.js'
+import {fakeMdxPath} from './path.js'
+import {mdxToJsx, unistPositionToTextSpan} from './utils.js'
+
+/**
+ * @param {string} fileName
+ * @returns {fileName is `${string}.mdx`} Whether or not the filename contains MDX.
+ */
+function isMdx(fileName) {
+  return fileName.endsWith('.mdx')
+}
+
+/**
+ * Assert that a file isn’t MDX.
+ *
+ * @param {string} fileName The file name to check.
+ * @param {string} fn The name of the function.
+ */
+function assertNotMdx(fileName, fn) {
+  if (isMdx(fileName)) {
+    throw new Error(`${fn} is not supported for MDX files`)
+  }
+}
+
+/**
+ * @param {string} fileName
+ * @param {MDXSnapshot | undefined} snapshot
+ * @param {TextSpan} textSpan
+ */
+function patchTextSpan(fileName, snapshot, textSpan) {
+  if (snapshot && isMdx(fileName)) {
+    textSpan.start = snapshot.getRealPosition(textSpan.start)
+  }
+}
+
+/**
+ * @param {string} fileName
+ * @param {MDXSnapshot} snapshot
+ * @param {NavigationBarItem} item
+ */
+function patchNavigationBarItem(fileName, snapshot, item) {
+  for (const span of item.spans) {
+    patchTextSpan(fileName, snapshot, span)
+  }
+
+  for (const child of item.childItems) {
+    patchNavigationBarItem(fileName, snapshot, child)
+  }
+}
+
+/**
+ * @param {import('typescript')} ts
+ * @param {LanguageServiceHost} host
+ * @param {PluggableList} [plugins]
+ * @returns {LanguageService} XXX
+ */
+export function createMDXLanguageService(ts, host, plugins) {
```

Comment:
```suggestion
export function createMdxLanguageService(ts, host, plugins) {
```

IMO further words in camelcase, even when they are normally uppercase in prose, have to be lowercased